mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -438,8 +443,14 @@ ModuleImport::processAliasScopeMetadata(const llvm::MDNode *node) {
     if (aliasDomain->getNumOperands() >= 2)
       if (auto *operand = dyn_cast<llvm::MDString>(aliasDomain->getOperand(1)))
         description = builder.getStringAttr(operand->getString());
-    return builder.getAttr<AliasScopeDomainAttr>(
-        DistinctAttr::create(builder.getUnitAttr()), description);
+    Attribute idAttr;
+    if (verifySelfRef(aliasDomain)) {
+      idAttr = DistinctAttr::create(builder.getUnitAttr());
+    } else {
+      auto name = cast<llvm::MDString>(aliasDomain->getOperand(0));
+      idAttr = builder.getStringAttr(name->getString());
+    }
```

Comment:
Maybe consider to also move this to a lambda, as this is exact snippet is duplicated.